packages/playwright-core/src/server/browserContext.ts

```diff
@@ -358,29 +359,30 @@ export abstract class BrowserContext extends SdkObject {
     this._timeoutSettings.setDefaultTimeout(timeout);
   }
 
-  async _loadDefaultContextAsIs(progress: Progress): Promise<Page[]> {
-    if (!this.pages().length) {
+  async _loadDefaultContextAsIs(progress: Progress): Promise<Page> {
+    if (!this.pagesOrErrors().length) {
       const waitForEvent = helper.waitForEvent(progress, this, BrowserContext.Events.Page);
       progress.cleanupWhenAborted(() => waitForEvent.dispose);
-      const page = (await waitForEvent.promise) as Page;
-      if (page._pageIsError)
-        throw page._pageIsError;
+      const page = await waitForEvent.promise as Page;
+      const pageOrError = await page._delegate.pageOrError();
+      if (pageOrError instanceof Error)
+        throw pageOrError;
     }
-    const pages = this.pages();
-    if (pages[0]._pageIsError)
-      throw pages[0]._pageIsError;
-    await pages[0].mainFrame()._waitForLoadState(progress, 'load');
-    return pages;
+    const pageOrError = await this.pagesOrErrors()[0];
+    if (pageOrError instanceof Error)
```

Comment:
Can you restructure the code so that `if (pageOrError instanceof Error)` performed just once?